### ISSUE TITLE
SSH and HTTP servers not working (likely due to a recent update)

### ISSUE THREAD

**Issue by JQ555888** [ish-app/ish#1350]
iSH reports "Bad system call" when a client tries to connect to an SSH/HTTP or other types of web server running on the device, resulting in an abruptly stopped process.

Locating the error with dmesg gives "missing syscall 364"

Steps to replicate:
- Run a python web server "python3 -m http.server" or start an ssh server "sshd -d -p [some high port]"
- Client tries to connect
- Process killed with last message Bad system call

**Comment by alexberry** [comment]
You should specify what version of ish and iOS you're using with your bug report, especially if you think an update broke things.

**Comment by 62f** [comment]
While attempting to create a workaround with tinyssh,
```missing syscall 151```
Attempt was made using build 132 because with 133, hiding the touch-kb during an IOERROR made it unretrievable.

**Comment by jaclu** [comment]
I can't comment on the HTTP issue, but as for SSH I don't seem to have any issue running an sshd either on a high ports or 22 with the current TestFlight version - Apr 10 2021 06:17:54 build and current iOS version.
I have been running it consistently for the last couple of months without noticing any failures to start. Granted every now and then the ssh session crashes, mostly due to iSH crashing, but a few times iSH survived but lost network.

**Comment by 62f** [comment]
TF's Apr. 10 release is iSH 1.1.2(135)
If iSH is crashing in-session there will be created an unsymbolicated file located in settings>privacy>analytics which can be both pasted here into a secret gist.GitHub.com or better yet shared with one of the devs ie. at the discord server. Especially useful info to forward since you're not on the very old AppStore release.

**Comment by gitdev-bash** [comment]
was on iSH 1.1.2(137) and iSH 1.1.2(135). I have never had an issue with "python.http" on both builds. But i cant say anything about "sshd" since i don't use the ssh server. Might be something special in your case.

**Comment by 62f** [comment]
> Was on iSH 1.1.2(137) and iSH 1.1.2(135). I have never had an issue with "python.http" on both builds. But i cant say anything about "sshd" since i don't use the ssh server. Might be something special in your case.

1 reason for bug-unreproductability: to be on a different Alpine release than the OP. What's described = having a workaround although the useful particulars of her functional configuration, @gitdev-bash left unstated.

**Comment by stheno** [comment]
app version 91 ssh and apache2 work fine for me. Not tried a python web server.
IOS 14.1 Ipad pro gen 1 9.7

**Comment by 62f** [comment]
> app version 91 ssh and apache2 work fine for me. Not tried a python web server.
> 
> IOS 14.1 Ipad pro gen 1 9.7

@stheno Traversing from TestFlight to AppStore release would be a niche-like downgrade. Likewise, iOS downgrades are impossible but not likely to solve anything anyway.
What info would be slightly more related is the version of Alpine which you'd used for this unbridled-success, even though an inexact configuration mock-up is mostly implemented as a temporary workaround; one with a high investment cost per issue.

**Comment by stheno** [comment]
So what exactly are you saying/asking for? I have not upgraded my iPad and have no intentions to. The version of alpine I used are completely out of box base that come with it when it is first installed.

**Comment by 62f** [comment]
@stheno your proposed workaround to this particular issue the OP is having with SSH, requires which of the following:
<details>
1)downgrading their firmware
2)installing an ancient (AppStore)!release in place of the OP's currently-installed daily build
3)installing an unrelated apk (Apache/2)
4)all in a totally different/specific model of iDevice</details>
?

Just interpreting any one of YOUR proposed changes to be steps of a solution could usher in a service tier whose scope is radically extreme(undoable), and even though it direction is unlikely to result in a fix, attempting to shoehorn into #1 may require, from yourself, providing advanced descriptions of modification far beyond the standard scope of many open-source GitHub projects, eg. even iSH;
Are YOU up to making that level of contributions?

**Comment by stheno** [comment]
I merely stated what I have that works. I didn't propose a darn thing.
Wow, you must be great at parties.

**Comment by 62f** [comment]
@stheno I behave quite differently  on social media outlets. 2) This is not a social media outlet. 3) Don't ever update your systems to become a clone of the OP's machine, or it too will also break.

**Comment by JQ555888** [comment]
My apologies for the late reply! It’s been a tiring month and I’ve just got the time to revisit this issue.

After several attempts of reinstalling the app and having the same issue happening on my other iOS devices, it turns out @62f is right indeed. I followed some old documentation that resulted me on the latest Alpine branch rather than the suggested one (I think it was 3.11) as implied in the iSH wiki page that basically tells you how to manually add the right Alpine repos. I thought it might be a good idea to stay on the latest for security reasons but I was being stupid and didn’t realize it could potentially break things. I am one of the testers from the early iSH days and I thought to myself when I did it I’d need to manually correct the repos as I migrate to the App Store version but obviously I made a mistake there. It was especially confusing as another iSH install of mine from the early days suffers from the same issue migrating from TestFlight to the App Store version (I effectively made a tar ball of the whole FS manually and applied it back since the device is jailbroken, the reason being that in doing so I hoped to evade the App Store mitigations). All in all, I was dumb and it’s all been my bad!

Many thanks to every single person who tried to help and contributed to the discussion! I wouldn’t have located the issue without all the suggestions. Closing this issue since it’s not strictly one that requires a bug fix.
